temoa/core/config.py

```
@@ -302,6 +302,12 @@ def build_config(config_file: Path, output_path: Path, silent: bool = False) ->
         else:
             raise SolverNotAvailableError('No solver name specified in the configuration.')
 
+        if data.get('solver_name') == 'appsi_highs' and data.get('save_duals', False):
+            raise ValueError(
+                'save_duals is not supported with appsi_highs (it does not expose duals via the '
+                'APPSI interface). Disable save_duals or choose a different solver.'
+            )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)